### PR TITLE
SPUI-422 > Update PinScreen (cold boot verification and lockTimer)

### PR DESCRIPTION
**Pull request by eli1stark**

### CODE REVIEW

File: lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart [hunk 1]
```diff
@@ -20,6 +25,7 @@ import 'pin_screen_union.dart';
 const pinLength = 4;
 const pinBoxAnimationDuration = Duration(milliseconds: 800);
 const pinBoxErrorDuration = Duration(milliseconds: 400);
+const defaultEnterPinAttempts = 5;
```
Comment (VovaMedvediev) [inline lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart:28]: move to remote config?

File: lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart [hunk 2]
```diff
@@ -119,22 +136,40 @@ class PinScreenNotifier extends StateNotifier<PinScreenState> {
           confirmPin: () => _confirmPinFlow(),
         );
       },
+      verification: () {
+        state.screenUnion.when(
+          enterPin: () => _enterPinFlow(),
+          newPin: () {}, // not needed
```
Comment (VovaMedvediev) [inline lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart:142]: maybe we can add default values to union or null values in order to get rid of such code

File: lib/shared/features/pin_screen/view/pin_screen.dart [hunk 3]
```diff
@@ -41,56 +45,96 @@ class PinScreen extends HookWidget {
...
-              child: Row(
-                mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+    Function()? onbackButton;
+    String? header;
+
+    if (union == const Verification()) {
+      onbackButton = () => logoutN.logout();
+      header = 'Simple';
+    } else if (cannotLeave) {
+      onbackButton = null;
```
Comment (VovaMedvediev) [inline lib/shared/features/pin_screen/view/pin_screen.dart:60]: it can be default value too (maybe)

Comment (eli1stark) [reply lib/shared/features/pin_screen/view/pin_screen.dart:60]: right

File: lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart (hunk 1 above)
Comment (eli1stark) [reply lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart:28]: will add the task to backlog (can't do this right away). Update time is 1- 2 days

File: lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart (hunk 2 above)
Comment (eli1stark) [reply lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart:142]: I'll look into it

File: lib/shared/features/pin_screen/view/pin_screen.dart (hunk 3 above)
Comment (eli1stark) [reply lib/shared/features/pin_screen/view/pin_screen.dart:61]:
```

    if (union == const Verification()) {
      onbackButton = () => logoutN.logout();
      header = 'Simple';
    } else if (cannotLeave) {
    } else {
      onbackButton = () => Navigator.pop(context);
      header = pin.screenHeader;
    }
```

looks weird, leaving as it was

File: lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart (hunk 2 above)
Comment (eli1stark) [reply lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart:141]:
```dart
disable: () {
        state.screenUnion.maybeWhen(
          enterPin: () => _enterPinFlow(),
          orElse: () {},
        );
      },
      enable: () {
        state.screenUnion.when(
          enterPin: () {}, // not needed
          newPin: () => _newPinFlow(),
          confirmPin: () => _confirmPinFlow(),
        );
      },
      verification: () {
        state.screenUnion.maybeWhen(
          enterPin: () => _enterPinFlow(),
          orElse: () {},
        );
      },
```

Comment (eli1stark) [reply lib/shared/features/pin_screen/notifier/pin_screen_notifier.dart:141]: good?

**Pull request merged**